### PR TITLE
[fix] Pricing rule for pos

### PR DESCRIPTION
**Pull request by rohitwaghchaure**

### CODE REVIEW

File: erpnext/accounts/page/pos/pos.js [hunk 1]
```diff
@@ -1779,6 +1779,13 @@ erpnext.pos.PointOfSale = erpnext.taxes_and_totals.extend({
 		})
 	},
 
+	validate_item_condition: function (data, item) {
+		var apply_on = frappe.model.scrub(data.apply_on);
+
+		return (data.apply_on == 'Item Group'
```
Comment (nabinhait) [inline erpnext/accounts/page/pos/pos.js]: IG 0
    - IG 01
        - IG 011
        - IG 012
    - IG 02

For the above Item group tree, if an item is belong to IG 012, the the pricing rule applicable to IG0 and IG01 should be evaluated.

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/accounts/page/pos/pos.js (hunk 1 above)
Comment (rohitwaghchaure) [inline erpnext/accounts/page/pos/pos.js]: Fixed

File: erpnext/accounts/doctype/sales_invoice/pos.py [hunk 2]
```diff
@@ -147,21 +147,26 @@ def get_items_list(pos_profile):
 		""".format(cond=cond), tuple(item_groups), as_dict=1)
 
 def get_item_group(pos_profile):
```
Comment (nabinhait) [inline erpnext/accounts/doctype/sales_invoice/pos.py]: get_item_groups ?

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/accounts/doctype/sales_invoice/pos.py (hunk 2 above)
Comment (rohitwaghchaure) [inline erpnext/accounts/doctype/sales_invoice/pos.py]: To get the list of item groups defined in the POS profile or all item groups

Comment (nabinhait) [inline erpnext/accounts/doctype/sales_invoice/pos.py]: I am just suggesting to rename the function

Comment (rohitwaghchaure) [inline erpnext/accounts/doctype/sales_invoice/pos.py]: Fixed

**Review by nabinhait — APPROVED** [review]

**Review by nabinhait — APPROVED** [review]

**Pull request merged**